app/src/main/kotlin/com/wire/android/ui/home/conversations/details/metadata/EditConversationMetadataViewModel.kt

```diff
@@ -0,0 +1,93 @@
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.id.QualifiedIdMapper
+import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
+import com.wire.kalium.logic.feature.conversation.RenameConversationUseCase
+import com.wire.kalium.logic.feature.conversation.RenamingResult
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@HiltViewModel
+class EditConversationMetadataViewModel @Inject constructor(
+    private val navigationManager: NavigationManager,
+    private val dispatcher: DispatcherProvider,
+    private val observeConversationDetails: ObserveConversationDetailsUseCase,
+    private val renameConversation: RenameConversationUseCase,
+    savedStateHandle: SavedStateHandle,
+    qualifiedIdMapper: QualifiedIdMapper
+) : ViewModel() {
+
+    private val conversationId: QualifiedID = qualifiedIdMapper.fromStringToQualifiedID(
+        savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)!!
+    )
+
+    var editConversationState by mutableStateOf(GroupMetadataState(mode = GroupNameMode.EDITION))
+        private set
+
+    init {
+        observeConversationDetails()
+    }
+
+    private fun observeConversationDetails() {
+        viewModelScope.launch {
+            observeConversationDetails(conversationId)
+                .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>()
+                .map { it.conversationDetails }
+                .filterIsInstance<ConversationDetails.Group>()
+                .distinctUntilChanged()
+                .flowOn(dispatcher.io())
+                .shareIn(this, SharingStarted.WhileSubscribed(), 1)
+                .collectLatest {
+                    editConversationState = editConversationState.copy(
+                        groupName = TextFieldValue(it.conversation.name.orEmpty()),
+                    )
+                }
+        }
+    }
+
+    fun onGroupNameChange(newText: TextFieldValue) {
+        editConversationState = GroupNameValidator.onGroupNameChange(newText, editConversationState)
+    }
+
+    fun onGroupNameErrorAnimated() {
+        editConversationState = GroupNameValidator.onGroupNameErrorAnimated(editConversationState)
+    }
+
+    fun saveNewGroupName() {
+        viewModelScope.launch {
+            when (withContext(dispatcher.io()) { renameConversation(conversationId, editConversationState.groupName.text) }) {
+                is RenamingResult.Failure -> appLogger.e("Failure changing name for conversation")
+                is RenamingResult.Success -> navigateBack()
```

Comment:
I think we should pass some value in `navigateBack` to show success snackbar in previous screen